### PR TITLE
Feature/upload clinical submission

### PR DESCRIPTION
**Pull request by jaserud**
**Description of changes**

Added schema and service for upload clinical submission. The logic is very similar to registration upload just for multiple files and clinicalTypes.

### CODE REVIEW

File: schemas/Clinical/index.js [hunk 1]
```diff
@@ -76,6 +76,40 @@ const typeDefs = gql`
     donorId: String!
     specimenId: String!
   }
+  
+  type ClinicalSubmissionData @cost(complexity: 10) {
+    id: ID    
+    state: String
```
Comment (joneubank) [inline schemas/Clinical/index.js]: If there is a known list of possible states, lets use an enum with those values.

File: schemas/Clinical/index.js [hunk 2]
```diff
@@ -76,6 +76,40 @@ const typeDefs = gql`
...
+    version: String
```
Comment (joneubank) [inline schemas/Clinical/index.js]: Are the versions integers?

File: schemas/Clinical/index.js [hunk 3]
```diff
@@ -178,6 +220,59 @@ const convertRegistrationDataToGql = data => {
   };
 };
 
+const convertClinicalSubmissionDataToGql = (data) => {
+  const submission = get(data, "submission", {});
+  const schemaErrors = get(data, "errors", {});
+  // convert clinical entities for gql
+  const clinicalEntities = [];
+  for (var clinicalType in submission.clinicalEntities) {
```
Comment (joneubank) [inline schemas/Clinical/index.js]: don't use `var` here, introduces clinicalType to the global scope. use `let` instead.

File: schemas/Clinical/index.js [hunk 4]
```diff
@@ -178,6 +220,59 @@ const convertRegistrationDataToGql = data => {
...
+    clinicalEntities.push(convertClinicalSubmissionEntityToGql(clinicalType, submission.clinicalEntities[clinicalType]));
+  }
+  // collect schema errors for each entity in dataErrors (not sure if this is OK??)
+  for (var clinicalType in schemaErrors) {
```
Comment (joneubank) [inline schemas/Clinical/index.js]: let instead of var

File: schemas/Clinical/index.js [hunk 5]
```diff
@@ -178,6 +220,59 @@ const convertRegistrationDataToGql = data => {
...
+    console.log("Curent entity:" + clinicalType);
```
Comment (joneubank) [inline schemas/Clinical/index.js]: remove console log

File: schemas/Clinical/index.js [hunk 6]
```diff
@@ -235,6 +330,24 @@ const resolvers = {
...
+    uploadClinicalSubmissions: async (obj, args, context, info) => {
+      const { Authorization, egoToken } = context;
+      const { shortName, clinicalFiles } = args;      
+
+      // see reason in uploadRegistration
+      if (!TokenUtils.canWriteSomeProgramData(egoToken)) {
+        throw new AuthenticationError('User is not authorized to write data');
+      }
+
+      var filesMap = {};
```
Comment (joneubank) [inline schemas/Clinical/index.js]: use const here

File: schemas/Clinical/index.js (hunk 2 above)
Comment (jaserud) [reply schemas/Clinical/index.js]: the version is a uuid

File: schemas/Clinical/index.js (hunk 1 above)
Comment (jaserud) [reply schemas/Clinical/index.js]: there is, I will add it then.

**Pull request merged**